incubator/hnc/test/e2e/demo_test.go

```diff
@@ -61,4 +61,23 @@ var _ = Describe("Demo", func() {
 		RunShouldContain(nsTeamB+"-wizard", 5, "kubectl -n", nsService1, "get roles")
 		RunShouldNotContain(nsTeamA+"-wizard", 5, "kubectl -n", nsService1, "get roles")
 	})
+
+	It("Should propagate different types", func(){
+		MustRun("kubectl create ns", nsOrg)
+		MustRun("kubectl hns create", nsTeamA, "-n", nsOrg)
+		MustRun("kubectl hns create", nsTeamB, "-n", nsOrg)
+		MustRun("kubectl create ns", nsService1)
+		MustRun("kubectl hns set", nsService1, "--parent", nsTeamB)
+		MustRun("kubectl -n", nsTeamB, "create secret generic my-creds --from-literal=password=iamteamb")
+		// secret does not show up in service-1 because we haven’t configured HNC to propagate secrets in HNCConfiguration.
+		RunShouldNotContain("my-creds", 5, "kubectl -n", nsService1, "get secrets")
+		MustRun("kubectl hns config set-type --apiVersion v1 --kind Secret propagate")
+		MustRun("kubectl get hncconfiguration config -oyaml")
+		RunShouldContain("my-creds", 5, "kubectl -n", nsService1, "get secrets")
```

Comment:
I'd have thought that a 2s timeout would be enough here, is that not true?

Reply:
2s is enough, I changed them all to 2s. I thought 5s might make it more stable, and for any command that takes less than 5s they will just return early